src/main/java/org/opensearch/ad/transport/AnomalyResultTransportAction.java

```diff
@@ -1081,7 +1129,7 @@ private Optional<Exception> coldStartIfNoCheckPoint(AnomalyDetector detector) {
             }
         }, exception -> {
             Throwable cause = ExceptionsHelper.unwrapCause(exception);
-            if (cause instanceof IndexNotFoundException) {
+            if (cause instanceof OpenSearchStatusException) {
```

Comment:
Same as above for the exception message

Reply:
I can change this to check if the `cause` message contains the string `type=index_not_found_exception` instead

Reply:
Let's do that